### ISSUE TITLE
openwrt 19.07.1 x86-64下编译通不过, 求教

### ISSUE THREAD

**Issue by blog2i2j** [wltc2005/openwrt-redsocks2#1]
您好, 我用openwrt sdk下编译您的这个redsocks2, 但编译通不过, 想请教一下如何解决, 谢谢!

x86_64-openwrt-linux-musl-gcc -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -iremap/home/op/openwrt-sdk-19.07.1-x86-64_gcc-7.5.0_musl.Linux-x86_64/build_dir/target-x86_64_musl/redsocks2//redsocks2-0.67:redsocks2-0.67 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -I/home/op/openwrt-sdk-19.07.1-x86-64_gcc-7.5.0_musl.Linux-x86_64/staging_dir/target-x86_64_musl/usr/include -I/home/op/openwrt-sdk-19.07.1-x86-64_gcc-7.5.0_musl.Linux-x86_64/staging_dir/target-x86_64_musl/include -I/home/op/openwrt-sdk-19.07.1-x86-64_gcc-7.5.0_musl.Linux-x86_64/staging_dir/toolchain-x86_64_gcc-7.5.0_musl/usr/include -I/home/op/openwrt-sdk-19.07.1-x86-64_gcc-7.5.0_musl.Linux-x86_64/staging_dir/toolchain-x86_64_gcc-7.5.0_musl/include/fortify -I/home/op/openwrt-sdk-19.07.1-x86-64_gcc-7.5.0_musl.Linux-x86_64/staging_dir/toolchain-x86_64_gcc-7.5.0_musl/include  -fPIC -O3 -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wall -std=c99 -D_XOPEN_SOURCE=600 -DUSE_CRYPTO_OPENSSL   -c -o encrypt.o encrypt.c
In file included from encrypt.c:50:0:
encrypt.h:92:18: error: field 'evp' has incomplete type
     cipher_evp_t evp;
                  ^~~
encrypt.c: In function 'enc_key_init':
encrypt.c:919:17: error: storage size of 'cipher_info' isn't known
     cipher_kt_t cipher_info;
                 ^~~~~~~~~~~
encrypt.c:919:17: warning: unused variable 'cipher_info' [-Wunused-variable]
At top level:
encrypt.c:146:12: warning: 'crypto_stream_xor_ic' defined but not used [-Wunused-function]
 static int crypto_stream_xor_ic(uint8_t *c, const uint8_t *m, uint64_t mlen,
            ^~~~~~~~~~~~~~~~~~~~
encrypt.c:139:18: warning: 'supported_ciphers_key_size' defined but not used [-Wunused-const-variable=]
 static const int supported_ciphers_key_size[] =
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
encrypt.c:134:18: warning: 'supported_ciphers_iv_size' defined but not used [-Wunused-const-variable=]
 static const int supported_ciphers_iv_size[] =
                  ^~~~~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [<builtin>: encrypt.o] Error 1

**Comment by wltc2005** [comment]
19.0x依赖的库都升级了，所以要改几个地方。我上次编译快半年前的事了我忘了改的是哪了。

**Comment by blog2i2j** [comment]
谢谢回复, 搜索了一番搞定了, 是OpenSSL库升级了, 原先的版本与库不兼容.
修改encrypt.c和encrypt.h就可以了.
修改的内容我放在这里了, 不会做补丁啊

https://github.com/blog2i2j/semigodking.._..redsocks/commit/cfd608747af49a9f56aaa029b7fab8ede15878c5
https://github.com/blog2i2j/semigodking.._..redsocks/commit/1e39d749bd1be153acf7383ae26548f6da5e67b1

**Comment by blog2i2j** [comment]
发现你把原版的fork过来后remove ss, 这样是可以搞定的.

**Comment by kousyougi** [comment]
我也遇到這個問題
請問要怎麼remove ss?
我只查到要加這行 DISABLE_SHADOWSOCKS=true ，但不知道該加到哪裡。
謝謝

**Comment by blog2i2j** [comment]
make package/resocks2/compile DISABLE_SHADOWSOCKS=true

**Comment by wltc2005** [comment]
我当时好像是把openwrt-redsocks2/Makefile中的
URL:=https://github.com/semigodking/redsocks
这一行地址改到自己已经删除ss的fork上
https://github.com/wltc2005/redsocks

**Comment by blog2i2j** [comment]
是的, redsocks2的作者把你的disable ss 给update到它的代码里了, 增加一个开关DISABLE_SHADOWSOCKS, 把这个开关打开就是跟你的代码一样了.
